shared_timed_mutex not available on older versions of MacOS

```
../include/deal.II/base/thread_local_storage.h:253:10: error: 'shared_timed_mutex' is unavailable: introduced in macOS 10.12
    std::shared_timed_mutex insertion_mutex;
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/shared_mutex:202:58: note: 'shared_timed_mutex' has been explicitly marked unavailable here
class _LIBCPP_TYPE_VIS _LIBCPP_AVAILABILITY_SHARED_MUTEX shared_timed_mutex
```

I just rolled back to version 9.2, which is OK for now. @tamiko I think that you introduced this change a short while ago. Is there an easy(ish) workaround for this? I know that my OS is old (circa Sept 2015), so if the answer is "no" then I recognise that I might just need to upgrade everything. I wonder, though, if there are users that might hit the same issue.

--- a/cmake/setup_compiler_flags_gnu.cmake
+++ b/cmake/setup_compiler_flags_gnu.cmake
@@ -36,6 +36,18 @@ IF( CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND
     )
 ENDIF()
 
+# Correspondence between AppleClang version and upstream Clang version:
+# https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)
+IF (POLICY CMP0025)
+  IF( CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND
+      CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9.0" )
+    MESSAGE(WARNING "\n"
+      "deal.II requires support for features of C++14 that are not present in\n"
+      "versions of AppleClang prior to 9.0."
+      )
+  ENDIF()
+ENDIF()
+
 
 ########################
 #                      #

--- a/cmake/setup_compiler_flags_gnu.cmake
+++ b/cmake/setup_compiler_flags_gnu.cmake
@@ -21,10 +21,10 @@
 #
 
 IF( CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9" )
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0" )
   MESSAGE(FATAL_ERROR "\n"
     "deal.II requires support for features of C++14 that are not present in\n"
-    "versions of GCC prior to 4.9."
+    "versions of GCC prior to 5.0."
     )
 ENDIF()
 

--- a/cmake/setup_compiler_flags_gnu.cmake
+++ b/cmake/setup_compiler_flags_gnu.cmake
@@ -22,7 +22,7 @@
 
 IF( CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND
     CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9" )
-  MESSAGE(WARNING "\n"
+  MESSAGE(FATAL_ERROR "\n"
     "deal.II requires support for features of C++14 that are not present in\n"
     "versions of GCC prior to 4.9."
     )
@@ -30,7 +30,7 @@ ENDIF()
 
 IF( CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND
     CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.0" )
-  MESSAGE(WARNING "\n"
+  MESSAGE(FATAL_ERROR "\n"
     "deal.II requires support for features of C++14 that are not present in\n"
     "versions of Clang prior to 4.0."
     )
@@ -41,7 +41,7 @@ ENDIF()
 IF (POLICY CMP0025)
   IF( CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND
       CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9.0" )
-    MESSAGE(WARNING "\n"
+    MESSAGE(FATAL_ERROR "\n"
       "deal.II requires support for features of C++14 that are not present in\n"
       "versions of AppleClang prior to 9.0."
       )

--- a/doc/readme.html
+++ b/doc/readme.html
@@ -63,7 +63,8 @@
     </p>
     <ul>
         <li>GNU/Linux: GCC version 4.9 or later; Clang version 4.0 or later; ICC versions 15 or later</li>
-        <li>Mac OS X: GCC version 4.9 or later; Clang version 4.0 or later. Please see the <a href="https://github.com/dealii/dealii/wiki/MacOSX" target="_top">deal.II Wiki</a> for installation instructions.</li>
+        <li>Mac OS X: GCC version 4.9 or later; Clang version 4.0 or later; <a href="https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)">Apple Clang version 9.0</a>  or later. 
+            Please see the <a href="https://github.com/dealii/dealii/wiki/MacOSX" target="_top">deal.II Wiki</a> for installation instructions.</li>
         <li>Windows: experimental support for Visual Studio 2017. Please have a look at the
             <a href="https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions#can-i-use-dealii-on-a-windows-platform">
       FAQ</a> and at the <a href="https://github.com/dealii/dealii/wiki/Windows" target="_top">deal.II Wiki</a> for more information and alternative solutions.</li>

--- a/doc/readme.html
+++ b/doc/readme.html
@@ -63,7 +63,7 @@
     </p>
     <ul>
         <li>GNU/Linux: GCC version 5.0 or later; Clang version 4.0 or later; ICC versions 15 or later</li>
-        <li>Mac OS X: GCC version 5.0 or later; Clang version 4.0 or later; <a href="https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)">Apple Clang version 9.0</a>  or later. 
+        <li>Mac OS X: GCC version 5.0 or later; Clang version 4.0 or later; <a href="https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)">Apple Clang version 9.0</a>  or later.
             Please see the <a href="https://github.com/dealii/dealii/wiki/MacOSX" target="_top">deal.II Wiki</a> for installation instructions.</li>
         <li>Windows: experimental support for Visual Studio 2017. Please have a look at the
             <a href="https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions#can-i-use-dealii-on-a-windows-platform">

--- a/doc/readme.html
+++ b/doc/readme.html
@@ -62,8 +62,8 @@
         <acronym>deal.II</acronym> supports at least the following platforms:
     </p>
     <ul>
-        <li>GNU/Linux: GCC version 4.9 or later; Clang version 4.0 or later; ICC versions 15 or later</li>
-        <li>Mac OS X: GCC version 4.9 or later; Clang version 4.0 or later; <a href="https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)">Apple Clang version 9.0</a>  or later. 
+        <li>GNU/Linux: GCC version 5.0 or later; Clang version 4.0 or later; ICC versions 15 or later</li>
+        <li>Mac OS X: GCC version 5.0 or later; Clang version 4.0 or later; <a href="https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)">Apple Clang version 9.0</a>  or later. 
             Please see the <a href="https://github.com/dealii/dealii/wiki/MacOSX" target="_top">deal.II Wiki</a> for installation instructions.</li>
         <li>Windows: experimental support for Visual Studio 2017. Please have a look at the
             <a href="https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions#can-i-use-dealii-on-a-windows-platform">

